### PR TITLE
Refactor/region detail

### PR DESCRIPTION
**Pull request by mmumby**

### CODE REVIEW

File: frontend/src/components/dashboard/Dashboard.js [hunk 1]
```diff
@@ -284,6 +280,7 @@ const mapStateToProps = (state) => {
     pageData: getMinesPageData(state),
     mineStatusOptions: getMineStatusOptions(state),
     mineRegionOptions: getMineRegionOptions(state),
+    mineRegionHash: getMineRegionHash(state)
```
Comment (NWCalvank) [inline frontend/src/components/dashboard/Dashboard.js]: Trailing comma

File: frontend/src/components/mine/MineHeader.js [hunk 2]
```diff
@@ -55,17 +56,22 @@ class MineHeader extends Component {
...
     };
+
     this.props.openModal({
       props: { mineStatusOptions, mineRegionOptions, onSubmit, title, initialValues },
       content: modalConfig.MINE_RECORD,
       clearOnSubmit: false,
     });
   }
 
+  renderRegion = () => {
```
Comment (NWCalvank) [inline frontend/src/components/mine/MineHeader.js]: Guessing this was left by accident?

File: frontend/src/selectors/mineSelectors.js [hunk 3]
```diff
@@ -36,4 +36,11 @@ export const getCurrentPermitteeIds = createSelector(
     unique = [...new Set(permitteeIds)];
     return unique;
   }
-);
\ No newline at end of file
+);
+
+export const getMineRegionHash = createSelector([getMineRegionOptions], (regions) => {
+  return regions.reduce(
```
Comment (NWCalvank) [inline frontend/src/selectors/mineSelectors.js]: If we care, Airbnb linting rules would not like this immediate `return` in an arrow function

File: frontend/src/tests/components/dashboard/Dashboard.spec.js [hunk 4]
```diff
@@ -23,6 +23,7 @@ const setupReducerProps = () => {
   reducerProps.pageData = MOCK.PAGE_DATA;
   reducerProps.mineStatusOptions = MOCK.STATUS_OPTIONS.options;
   reducerProps.mineRegionOptions = MOCK.REGION_OPTIONS.options;
+  reducerProps.mineRegionHash = MOCK.REGION_HASH
```
Comment (NWCalvank) [inline frontend/src/tests/components/dashboard/Dashboard.spec.js]: Semicolon

File: frontend/src/tests/components/dashboard/MineList.spec.js [hunk 5]
```diff
@@ -8,6 +8,7 @@ const props = {};
 const setupProps = () => {
   props.mineIds = MOCK.MINES.mineIds;
   props.mines = MOCK.MINES.mines;
+  props.mineRegionHash = MOCK.REGION_HASH
```
Comment (NWCalvank) [inline frontend/src/tests/components/dashboard/MineList.spec.js]: Semicolon

File: frontend/src/tests/components/mine/MineHeader.spec.js [hunk 6]
```diff
@@ -17,6 +17,7 @@ const setupProps = () => {
   props.mine = MOCK.MINES.mines[MOCK.MINES.mineIds[0]];
   props.mineStatusOptions = MOCK.STATUS_OPTIONS.options;
   props.mineRegionOptions = MOCK.REGION_OPTIONS.options;
+  props.mineRegionHash = MOCK.REGION_HASH
```
Comment (NWCalvank) [inline frontend/src/tests/components/mine/MineHeader.spec.js]: Semicolon

File: frontend/src/tests/mocks/dataMocks.js [hunk 7]
```diff
@@ -22,24 +22,19 @@ export const MINE_RESPONSE = {
       {
         mine_name: "mine1",
         mine_no: "BLAH9091",
+        region_code: "NE"
```
Comment (NWCalvank) [inline frontend/src/tests/mocks/dataMocks.js]: Trailing commas

File: frontend/src/tests/mocks/dataMocks.js [hunk 8]
```diff
@@ -455,6 +435,14 @@ export const REGION_OPTIONS = {
   }
 ]};
 
+export const REGION_HASH = {
+  "SW": "South West",
```
Comment (NWCalvank) [inline frontend/src/tests/mocks/dataMocks.js]: I think the quotes are unnecessary here. Airbnb linting rules requests removing them if they're not needed

File: frontend/src/tests/selectors/mineSelectors.spec.js [hunk 9]
```diff
@@ -7,7 +7,8 @@ import {
   getMineGuid,
   getCurrentPermittees,
   getCurrentPermitteeIds,
-  getMinesPageData
+  getMinesPageData,
+  getMineRegionHash
```
Comment (NWCalvank) [inline frontend/src/tests/selectors/mineSelectors.spec.js]: Trailing comma

File: frontend/src/tests/selectors/mineSelectors.spec.js [hunk 10]
```diff
@@ -115,4 +116,10 @@ describe('mineSelectors', () => {
     const selected = getCurrentPermitteeIds.resultFunc(mines, mineGuid);
     expect(selected).toEqual(["1c7da2c4-10d5-4c9f-994a-96427aa0c69b"]);
   });
+
+  it('`getMineRegionHash` calls `mineReducer.getCurrentPermitteeIds`', () => {
+    mineRegionOptions = Mock.REGION_OPTIONS.options
```
Comment (NWCalvank) [inline frontend/src/tests/selectors/mineSelectors.spec.js]: Semicolon

File: migrations/etl/1_Mine_profile_detail_location.sql [hunk 11]
```diff
@@ -108,7 +108,7 @@ BEGIN
         create_timestamp    ,
         update_user         ,
         update_timestamp    ,
-        major               )
+        major_mine_ind      )
```
Comment (NWCalvank) [inline migrations/etl/1_Mine_profile_detail_location.sql]: Mind doing this on the other lists in this file?

File: python-backend/tests/constants.py [hunk 12]
```diff
@@ -55,8 +55,8 @@
 TEST_REGION_GUID = 'a0b98cb1-6e70-4c54-9395-1e8ccc867cd1'
 TEST_REGION_CODES = ['SW','SC','NW','NE','SE']
 TEST_REGION_CODE_DISPLAY_ORDER = [10,20,30,40,50]
-TEST_REGION_CODE_1 = 'SW'
-TEST_REGION_DESCRIPTION= 'TEST CODE'
+TEST_REGION_CODE = 'NE'
+TEST_REGION_DESCRIPTION= 'North East Region'
```
Comment (NWCalvank) [inline python-backend/tests/constants.py]: Do we still have a use for descriptions on the server? I guess I misunderstood, but I thought they were being removed. Or maybe they were just being returned to the client in a different way?

File: migrations/sql/V1.5__Alter_Mine_Region.sql [hunk 13]
```diff
@@ -0,0 +1,15 @@
...
+;
+
+ALTER TABLE mine_region_code RENAME region_code TO mine_region_code;
+
+ALTER TABLE mine_detail 
+    ADD mine_region character varying(2) NOT NULL,
+    ADD FOREIGN KEY(mine_region) REFERENCES mine_region_code(mine_region_code) 
+;
+
+
```
Comment (NWCalvank) [inline migrations/sql/V1.5__Alter_Mine_Region.sql]: Extra newline padding

File: python-backend/tests/constants.py (hunk 12 above)
Comment (VivianCheungFW) [reply python-backend/tests/constants.py]: We still have description in the mine_region_code table. But we removed it in one of the returned lists.

File: frontend/src/components/mine/MineHeader.js (hunk 2 above)
Comment (mmumby) [reply frontend/src/components/mine/MineHeader.js]: deleted. GOOOD CATCH

**Pull request closed without merging**